### ISSUE TITLE
add output for progress through geometry clean operations

### ISSUE THREAD

**Issue by Agent6-6-6** [robbievanleeuwen/section-properties#19]
Hi Robbie

**Is your feature request related to a problem? Please describe.**
When you are cleaning the geometry of a section with a large number of mesh elements, it can take a significant amount of time (like an hour+ for complex sections) with nothing being fed back to the user to indicate progress if you haven't used verbose=True.

There is currently no feedback on progress through this sometimes lengthy process if verbose=False is used, for example when you display the results for other procedures with (time_info=True) it gives you the time taken and what is being looked at as an analysis is stepped through. The geometry clean routine can take significantly longer than the other section property calculations, but there is no time feedback for this process currently.

**Describe the solution you'd like**
In its most basic form of timing each geometry clean subroutine it might look like below to provide some ongoing feedback to the user through the clean process in a similar way to all the other time_info code for other calculations. without needing to resort to verbose=True to see if its making progress.

```
if time_info:
            text = "--Merging close points..."
            solver.function_timer(text, self.zip_points)
            print("")
            text = "--Removing zero length facets..."
            solver.function_timer(text, self.remove_zero_length_facets)
            print("")
            text = "--Removing duplicate facets..."
            solver.function_timer(text, self.remove_duplicate_facets)
            print("")
            text = "--Removing overlapping facets..."
            solver.function_timer(text, self.remove_overlapping_facets)
            print("")
            text = "--Removing unused points..."
            solver.function_timer(text, self.remove_unused_points)
            print("")
            text = "--Rebuilding intersecting facets..."
            solver.function_timer(text, self.intersect_facets)
            print("")
        else:
            self.zip_points()
            self.remove_zero_length_facets()
            self.remove_duplicate_facets()
            self.remove_overlapping_facets()
            self.remove_unused_points()
            self.intersect_facets()
```
I could spend a small amount of time and follow through altering the required modules if you were in agreement with this as an addition and submit it as a pull request?

Thanks!

**Comment by connorferster** [comment]
@Agent6-6-6 

Ah! I know just what you mean! I gave up on it (waiting for geometry cleaning to complete) and that's when I felt resolved to complete the shapely preprocessor. My hope was to entirely eliminate that geometry cleaning step, instead relying on shapely to always provide valid geometry.

The one place I have kept it in is for the current calculation for the plastic properties. While this can likely be done entirely in shapely, for now I have left plastic properties calculation just as Robbie made it, which requires a relatively short cleanup process. 

Before you start on the revisions, I wanted to mention that my plan is to largely do away with geometry cleaning to (hopefully) prevent the need for long wait times.
